sockets.js

```diff
@@ -443,48 +483,55 @@ if (cluster.isMaster) {
 		sockets.clear();
 		channels.clear();
 		subchannels.clear();
+
 		app.close();
 		if (appssl) appssl.close();
+
+		// Let the server(s) finish closing.
+		setImmediate(() => process.exit(0));
 	});
 
 	// this is global so it can be hotpatched if necessary
 	let isTrustedProxyIp = Dnsbl.checker(Config.proxyip);
 	let socketCounter = 0;
 	server.on('connection', socket => {
-		if (!socket) {
-			// For reasons that are not entirely clear, SockJS sometimes triggers
-			// this event with a null `socket` argument.
-			return;
-		} else if (!socket.remoteAddress) {
-			// This condition occurs several times per day. It may be a SockJS bug.
+		// For reasons that are not entirely clear, SockJS sometimes triggers
+		// this event with a null `socket` argument.
+		if (!socket) return;
+
+		if (!socket.remoteAddress) {
+			// SockJS will sometimes fail to cache a connection's IP, protocol,
+			// and address sometimes. When this happens is likely dependent on
+			// the transport being used and what browser's running the client.
+			// TODO: look more into why this happens.
 			try {
 				socket.destroy();
 			} catch (e) {}
 			return;
 		}
 
-		let socketid = socket.id = '' + (++socketCounter);
+		let socketid = ++socketCounter;
 		sockets.set(socketid, socket);
 
-		if (isTrustedProxyIp(socket.remoteAddress)) {
+		let socketip = socket.remoteAddress;
```

Comment:
Are you really going to set a temporary variable because you use a member _twice_?

Reply:
It's not even a calculated value, it's literally just a property access...

Reply:
Oh, nevermind, I see what you're doing now. This is the "don't write to sockets" thing.